--- a/etc/init.d/S90lmd
+++ b/etc/init.d/S90lmd
@@ -8,6 +8,12 @@ PROG=/usr/bin/lmd

 case "$1" in
     start)
+        EXTENDED_CFG="/home/lava/printer_data/config/extended/extended2.cfg"
+        CAMERA_INTERNAL=$(/usr/local/bin/extended-config.py get "$EXTENDED_CFG" camera internal paxx12)
+        if [ "$CAMERA_INTERNAL" != "snapmaker" ]; then
+            echo "Internal camera is not set to 'snapmaker', not starting lmd."
+            exit 0
+        fi
         ulimit -c 102400
         start-stop-daemon -S -b -x "$PROG"
         ;;
